actionpack/lib/action_controller/metal/head.rb

```diff
@@ -17,8 +17,14 @@ module Head
     #
     # See Rack::Utils::SYMBOL_TO_STATUS_CODE for a full list of valid +status+ symbols.
     def head(status, options = {})
-      options, status = status, nil if status.is_a?(Hash)
-      status ||= options.delete(:status) || :ok
+      if status.is_a?(Hash)
+        msg = "Passing Hash as first parameter"
+        options, status = status, nil
+        status ||= options.delete(:status) || (
+          (msg += " and default status code") && :ok
```

Comment:
"Passing Hash as first parameter and default status code" doesn't make sense to me. What about `and the implicit :ok status has been deprecated.`?

It's okay to not have the code be completely DRY. I'd prefer making this more readable :smile: